src/replaceCustomValues.ts

```diff
@@ -694,3 +694,178 @@ export function replaceInjectablesWithSubmissionValues(
     hadAllInjectablesReplaced,
   }
 }
+
+/**
+ * Process a resource with injectable element values to turn a single resource
+ * (could be a single) into multiple resources. e.g.
+ * `"{ELEMENT:Children|Child_Name} {ELEMENT:Family_Name}"` with the following
+ * submission data:
+ *
+ * ```json
+ * {
+ *   "Family_Name": "Smith",
+ *   "Children": [
+ *     {
+ *       "Child_Name": "John"
+ *     },
+ *     {
+ *       "Child_Name": "Jane"
+ *     }
+ *   ]
+ * }
+ * ```
+ *
+ * Would result in the following resources:
+ *
+ * - `"John Smith"`
+ * - `"Jane Smith"`
+ *
+ * #### Example
+ *
+ * ```js
+ * const emailAddresses = processInjectableResource(
+ *   '{ELEMENT:People|Email_Address}',
+ *   {
+ *     People: [
+ *       {
+ *         Email_Address: 'user@oneblink.io',
+ *       },
+ *       {
+ *         Email_Address: 'admin@oneblink.io',
+ *       },
+ *     ],
+ *   },
+ *   [
+ *     {
+ *       id: '18dcd3e0-6e2f-462e-803b-e24562d9fa6d',
+ *       type: 'repeatableSet',
+ *       name: 'People',
+ *       label: 'People',
+ *       elements: [
+ *         {
+ *           id: 'd0902113-3f77-4070-adbd-ca3ae95ce091',
+ *           type: 'email',
+ *           name: 'Email_Address',
+ *           label: 'Email_Address',
+ *         },
+ *       ],
+ *     },
+ *   ],
+ *   (resource, submission, formElements) => {
+ *     const { text } = replaceInjectablesWithElementValues(resource, {
+ *       submission,
+ *       formElements,
+ *       // other options
+ *     })
+ *     return text
+ *   },
+ * )
+ * // emailAddresses === ["user@oneblink.io", "admin@oneblink.io"]
+ * ```
+ *
+ * @param resource The resource that contains properties that support injection
+ *   or a string
+ * @param submission The form submission data to process
+ * @param formElements The form elements to process
+ * @param injector A function to inject values, this allows custom formatters to
+ *   be used. Return `undefined` to prevent the injection from recursively
+ *   continuing.
+ * @param replacer An optional function to replace nested injectables when
+ *   creating multiple resources from repeatable sets. Only required if the
+ *   `resource` param is not a `string`.
+ * @returns
+ */
+export function processInjectableResource<T>(
+  resource: T,
+  submission: SubmissionTypes.S3SubmissionData['submission'],
+  formElements: FormTypes.FormElement[],
+  injector: (
+    resource: T,
+    submission: SubmissionTypes.S3SubmissionData['submission'],
+    formElements: FormTypes.FormElement[],
+  ) =>
+    | [injectedText: string, resourceKey: string, newResource: T]
+    | string
+    | undefined,
+  replacer: (resource: T, replaceAll: (resourceText: string) => string) => T = (
```

Comment:
And this be called something like `replaceNestedInjectables` or `replaceRepeatableSetChildrenInjectables`